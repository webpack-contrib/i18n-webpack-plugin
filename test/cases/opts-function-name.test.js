import processFile from '../cases.setup';

describe('options.functionName', () => {
  describe('the object notation', () => {
    let translated;

    beforeAll(() => {
      const translations = {
        key: 'translated key',
      };
      const options = {
        functionName: 'i18n',
      };

      return processFile('opts-function-name.code.js', translations, options)
        .then(({ file }) => {
          translated = require.requireActual(file);
        });
    });

    it('should return translated keys', () => {
      expect(translated).toMatchSnapshot();
    });
  });

  describe('the string notation', () => {
    let translated;

    beforeAll(() => {
      const translations = {
        key: 'translated key',
      };
      const options = 'i18n';

      return processFile('opts-function-name.code.js', translations, options)
        .then(({ file }) => {
          translated = require.requireActual(file);
        });
    });

    it('should return translated keys', () => {
      expect(translated).toMatchSnapshot();
    });
  });
});
