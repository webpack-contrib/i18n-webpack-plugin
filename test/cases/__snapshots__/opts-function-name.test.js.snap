// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options.functionName the object notation should return translated keys 1`] = `
Object {
  "key": "translated key",
}
`;

exports[`options.functionName the string notation should return translated keys 1`] = `
Object {
  "key": "translated key",
}
`;
